src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -1653,14 +1653,22 @@ void Instruction::execute() {
       }
       case Opcode::AArch64_FDIVR_ZPmZ_D: {  // fdivr zdn.d, pg/m, zdn.d, zm.d
         results[0] = sveLogicOpPredicated_3vecs<double>(
-            operands, VL_bits,
-            [](double x, double y) -> double { return (y / x); });
+            operands, VL_bits, [](double x, double y) -> double {
+              if (x == 0)
+                return std::nan(0);
+              else
+                return (y / x);
+            });
```

Comment:
I'm not a huge fan of the passed function being this complex. Is it worth us creating helper functions for fdiv and fdivr? We can use `enable_if` to ensure the given types are floating point and then check the size of the typename to decide if we should use `nan` or `nanf`

Reply:
In my opinion, adding a conditional and a return of a nan doesn't add much complexity so I don't think that point stands. Saying that, I'm by no means tied to this so happy to add a fdiv helper function. Given that fdivr is just fidv reversed, we could probably get away with one helper function and reverse the input operands.

Reply:
Or on the one helper function point, we could pass a bool in the template to denote whether its reversed?

Reply:
Complex was probably the wrong word --- it just looks a bit bloated perhaps?

I think implementing an fdiv helper would aid consistency with the rest of the execute switch case. Passing a bool to denote when operands are reversed is good - we do something similar for other helper functions